### PR TITLE
[WA-31]: details information for today

### PR DESCRIPTION
**Pull request by Anurov-Prihodko**

### CODE REVIEW

File: src/app/details/components/details-table/details-table.component.ts [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+  constructor() {}
 
-  constructor() { }
-
-  ngOnInit(): void {
-  }
+  ngOnInit(): void {}
 
+  @Input() dataDetails: DataDetailsWeather = {
+    temp_c: [],
```
Comment (dmitry-chirva) [inline src/app/details/components/details-table/details-table.component.ts:15]: Do you like this interface?

Comment (dmitry-chirva) [inline src/app/details/components/details-table/details-table.component.ts:15]: Do you like this interface?

File: src/app/details/details.component.ts [hunk 2]
```diff
@@ -27,5 +38,27 @@ export class DetailsComponent {
...
+        ({
+          temp_c,
+          feelslike_c,
+          wind_kph,
+          gust_kph,
+          cloud,
+          humidity,
+          pressure_mb,
+        }: DataDetailsWeather) => {
+          this.detailsWeatherData.temp_c = [...temp_c];
```
Comment (dmitry-chirva) [inline src/app/details/details.component.ts:54]: Why do we need these?

Comment (dmitry-chirva) [inline src/app/details/details.component.ts:54]: Why do we need these?

**Comment by dmitry-chirva** [comment]
Need work on this task

**Comment by dmitry-chirva** [comment]
Need work on this task

File: src/app/details/components/details-table/details-table.component.ts (hunk 1 above)
Comment (Anurov-Prihodko) [reply src/app/details/components/details-table/details-table.component.ts:15]: changed

File: src/app/details/details.component.ts (hunk 2 above)
Comment (Anurov-Prihodko) [reply src/app/details/details.component.ts:54]: changed

File: src/app/details/components/details-table/details-table.component.ts [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 export class DetailsTableComponent implements OnInit {
+  constructor() {}
 
-  constructor() { }
-
-  ngOnInit(): void {
-  }
+  ngOnInit(): void {}
 
+  @Input() dataDetails: DataDetailsWeather = {
```
Comment (dmitry-chirva) [inline src/app/details/components/details-table/details-table.component.ts:14]: interface DetailsWeather

File: src/app/details/components/details-table/details-table.component.ts [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
+import { DataDetailsWeather } from 'src/app/shared/interfaces/details-weather-data.interfaces';
 
 @Component({
   selector: 'app-details-table',
   templateUrl: './details-table.component.html',
-  styleUrls: ['./details-table.component.scss']
+  styleUrls: ['./details-table.component.scss'],
 })
 export class DetailsTableComponent implements OnInit {
+  constructor() {}
```
Comment (dmitry-chirva) [inline src/app/details/components/details-table/details-table.component.ts:10]: remove constructor and ngOnInit

File: src/app/details/components/details-table/details-table.component.ts [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
 
-  constructor() { }
-
-  ngOnInit(): void {
-  }
+  ngOnInit(): void {}
 
+  @Input() dataDetails: DataDetailsWeather = {
+    celsiusTemperature: [],
+    celsiusTemperatureFeelslike: [],
```
Comment (dmitry-chirva) [inline src/app/details/components/details-table/details-table.component.ts:16]: rename specific prefix like celsius

File: src/app/details/details.component.ts [hunk 6]
```diff
@@ -27,5 +38,21 @@ export class DetailsComponent {
...
       });
+
+    this.detailsService
+      .getDataForWeatherTable()
+      .subscribe((data: DataDetailsWeather[] | any) => {
+        if (!data) {
+          return;
+        }
+
+        this.detailsWeatherData.celsiusTemperature = data.temp_c;
```
Comment (dmitry-chirva) [inline src/app/details/details.component.ts:49]: transform(data: backendInterface): DataDetailsWeather {
  return  {
    celsiusTemperature: data.temp_c,
    celsiusTemperatureFeelslike: data.feelslike_c,
    windSpeed: [],
    windSpeedFeelslike: [],
    cloud: [],
    humidity: [],
    pressure: []
  }
}

File: src/app/shared/interfaces/forecast-info.interfaces.ts [hunk 7]
```diff
@@ -47,6 +47,9 @@ export interface Condition {
 }
 
 export interface Forecast {
+  current: any;
+  location: any;
+  forecast: any;
```
Comment (dmitry-chirva) [inline src/app/shared/interfaces/forecast-info.interfaces.ts:52]: remove any

File: src/core/api/details/transform-data-details.service.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+})
+export class TransformDataDetailsService {
+  private TO_METERS_PER_SEC = 0.278;
+
+  constructor() {}
+
+  getDataDetails(data: Forecast): Observable<ForecastData> {
+    return new Observable((obs) => {
+      obs.next(
+        data.forecast.forecastday[0].hour.forEach((v: Hour) => {
```
Comment (dmitry-chirva) [inline src/core/api/details/transform-data-details.service.ts:20]: rename v to day

File: src/app/details/components/details-table/details-table.component.ts (hunk 4 above)
Comment (Anurov-Prihodko) [reply src/app/details/components/details-table/details-table.component.ts:10]: Done

File: src/app/details/components/details-table/details-table.component.ts (hunk 3 above)
Comment (Anurov-Prihodko) [reply src/app/details/components/details-table/details-table.component.ts:14]: Done

File: src/core/api/details/transform-data-details.service.ts (hunk 8 above)
Comment (Anurov-Prihodko) [reply src/core/api/details/transform-data-details.service.ts:20]: Done

File: src/app/details/details.component.ts (hunk 6 above)
Comment (Anurov-Prihodko) [reply src/app/details/details.component.ts]: Done

File: src/app/shared/components/table/table.component.html [hunk 9]
```diff
@@ -1,6 +1,75 @@
-<table class="table">
-    <tr *ngFor="let column of columns">
-        <td>{{column.displayName}}</td>
-        <td *ngFor="let row of data">{{row[column.name]}}</td>
-    </tr>
+<table class="details-table">
```
Comment (dmitry-chirva) [inline src/app/shared/components/table/table.component.html:1]: ??

**Comment by dmitry-chirva** [comment]
Need refactoring

**Pull request merged**